### PR TITLE
Wavefront Service Broker for Cloud Foundry

### PR DESCRIPTION
**Pull request by vikramraman**
Service broker for Cloud Foundry

### CODE REVIEW

File: src/main/java/com/wavefront/cloudfoundry/servicebroker/service/ProxyServiceInstanceBindingService.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * Service for binding/unbinding apps to a proxy service instance.
+ *
+ * @author Vikram Raman
+ */
+@Service
+public class ProxyServiceInstanceBindingService implements ServiceInstanceBindingService {
+
+  private ProxyServiceInstanceBindingRepository repository;
```
Comment (panghy) [inline src/main/java/com/wavefront/cloudfoundry/servicebroker/service/ProxyServiceInstanceBindingService.java]: final also for proxyConfig

File: src/main/java/com/wavefront/cloudfoundry/servicebroker/service/ProxyServiceInstanceService.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * Instantiated proxy service instances provide a mechanism for third party apps
+ * to retrieve routing information for a deployed proxy application.
+ *
+ * @author Vikram Raman
+ */
+@Service
+public class ProxyServiceInstanceService implements ServiceInstanceService {
+
+  private ProxyServiceInstanceRepository repository;
```
Comment (panghy) [inline src/main/java/com/wavefront/cloudfoundry/servicebroker/service/ProxyServiceInstanceService.java]: final

File: manifest.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- name: wavefront-proxy-service-broker
+  instances: 1
+  memory: 1G
+  path: target/wavefront-cloudfoundry-broker-0.0.1-SNAPSHOT.jar
+  buildpack: https://github.com/cloudfoundry/java-buildpack.git
+  # change the route domain name accordingly
+  route:
+  - route: wavefront-proxy-service-broker.apps.wavefront.io
+  env:
+    SECURITY_USER_NAME: admin
```
Comment (panghy) [inline manifest.yml:12]: what is this username/password?

**Review by panghy — APPROVED** [review]
LGTM otherwise

File: manifest.yml (hunk 3 above)
Comment (vikramraman) [reply manifest.yml:12]: This is just a default username/password that is used for dev testing when we manually push this as an app. The CF framework auto generates these when registering a service broker as part of the tile.

**Comment by vikramraman** [comment]
@panghy Addressed your comments.

**Pull request merged**